### PR TITLE
Add missing javadocs nearby #1828 Tidy up code in entire Nearby package and write Javadocs for all Nearby classes and methods 

### LINKED ISSUE #1828: Tidy up code in entire Nearby package and write Javadocs for all Nearby classes and methods
As per #1683 : This will help allow new contributors to get started with this package, and will also provide a solid foundation for future Nearby improvements. If successful, we plan to iterate on this with other packages in the future.

### PR DESCRIPTION
**Pull request by neslihanturan**
**Description (required)**

Fixes #1828 Tidy up code in entire Nearby package and write Javadocs for all Nearby classes and methods 

What changes did you make and why?

Added missing javadocs in some Nearby classes.

**Tests performed (required)**

Tested betaDebug API 26

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java [hunk 1]
```diff
@@ -375,6 +373,10 @@ private void populatePlacesFromCustomLocation(NearbyController.NearbyPlacesInfo
         }
     }
 
+    /**
```
[the remark this answers is not in the document]
Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java]: Apparently I missed a whole class

[the remark this answers is not in the document]
Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java]: Added, thanks @domdomegg !

**Comment by domdomegg** [comment]
Seems fine if I rebase, but without rebasing unit testing fails at `UploadModelTest`

If Travis fails in the unit tests, might be worth rebasing.

```
io.reactivex.exceptions.OnErrorNotImplementedException: The mapper function returned a null value.
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.ConsumerSingleObserver.onError(ConsumerSingleObserver.java:47)
	at io.reactivex.internal.operators.single.SingleZipArray$ZipCoordinator.innerError(SingleZipArray.java:139)
	at io.reactivex.internal.operators.single.SingleZipArray$ZipSingleObserver.onError(SingleZipArray.java:177)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onError(SingleMap.java:69)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onError(SingleMap.java:69)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:60)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:56)
	at io.reactivex.Single.subscribe(Single.java:3394)
	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
	at io.reactivex.Single.subscribe(Single.java:3394)
	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
	at io.reactivex.Single.subscribe(Single.java:3394)
	at io.reactivex.internal.operators.single.SingleZipArray.subscribeActual(SingleZipArray.java:64)
	at io.reactivex.Single.subscribe(Single.java:3394)
	at io.reactivex.Single.subscribe(Single.java:3380)
	at io.reactivex.Single.subscribe(Single.java:3351)
	at fr.free.nrw.commons.upload.UploadModel.receiveDirect(UploadModel.java:146)
	at fr.free.nrw.commons.upload.UploadModelTest.isTopCardStateForDirectUpload(UploadModelTest.kt:226)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:131)
Caused by: java.lang.NullPointerException: The mapper function returned a null value.
	at io.reactivex.internal.functions.ObjectHelper.requireNonNull(ObjectHelper.java:39)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:57)
	... 49 more
```

**Comment by domdomegg** [comment]
Travis seems okay with unit tests. As unit tests are fine once rebased I'm happy.

Tested `2.9.0-debug-pr-2133~02ac4efaa` on `Samsung Galaxy S6` at API level `25`

**Pull request merged**